src/utils/apis/auth.js

```diff
@@ -0,0 +1,48 @@
+export const BASE_URL =
+  process.env.NODE_ENV === "production"
+    ? "https://api.haikukedoodle.com"
+    : "http://localhost:3001";
+
+const handleAuthResponse = (res) => {
+  if (!res.ok) {
+    return Promise.reject(`Error:${res.status}`);
+  } else {
+    return res.json();
+  }
+};
+
+export const signup = (
+  name,
+  email,
+  password,
+  isAnonymous,
+  counter,
+  counterTimeStamp,
+  counterMax
+) => {
+  return fetch(`${BASE_URL}/signup`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      name,
+      email,
+      password,
+      isAnonymous,
+      counter,
+      counterTimeStamp,
+      counterMax,
+    }),
+  }).then(handleAuthResponse);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(handleResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.

Reply:
hey gennady where should this function live? In helpers or in the api or auth file?